src/jsonapi.js

```diff
@@ -300,157 +377,199 @@ export const requireResource = (resourceType, endpoint = resourceType, {
+      );
+
+      return imm(newState)
+        .set("isCreating", state.isCreating - 1)
+        .value();
+    },
+
+    [API_CREATE_FAILED]: (state) => {
+      return imm(state)
+        .set(["isCreating"], state.isCreating - 1)
+        .value();
+    },
+
+    [API_WILL_READ]: (state) => {
+      return imm(state)
+        .set(["isReading"], state.isReading + 1)
+        .value();
+    },
+
+    [API_READ]: (state, { payload }) => {
+      const resources = (
+        Array.isArray(payload.data) ? payload.data : [payload.data]
+      ).concat(payload.included || []);
+
+      const newState = updateOrInsertResourcesIntoState(state, resources);
+      const finalState = addLinksToState(
+        newState,
+        payload.links,
+        payload.options
+      );
+
+      return imm(finalState)
+        .set("isReading", state.isReading - 1)
+        .value();
+    },
+
+    [API_READ_FAILED]: (state) => {
+      return imm(state)
+        .set(["isReading"], state.isReading - 1)
+        .value();
+    },
+
+    [API_WILL_UPDATE]: (state, { payload: resource }) => {
+      const { type, id } = resource;
+
+      const newState = ensureResourceTypeInState(state, type);
+
+      return setIsInvalidatingForExistingResource(
+        newState,
+        { type, id },
+        IS_UPDATING
+      )
+        .set("isUpdating", state.isUpdating + 1)
+        .value();
+    },
+
+    [API_UPDATED]: (state, { payload: resources }) => {
+      const entities = Array.isArray(resources.data)
+        ? resources.data
+        : [resources.data];
+
+      const newState = updateOrInsertResourcesIntoState(
+        state,
+        entities.concat(resources.included || [])
+      );
+
+      return imm(newState)
+        .set("isUpdating", state.isUpdating - 1)
+        .value();
+    },
+
+    [API_UPDATE_FAILED]: (state, { payload: { resource } }) => {
+      const { type, id } = resource;
+
+      return setIsInvalidatingForExistingResource(
+        state,
+        { type, id },
+        IS_UPDATING
+      )
+        .set("isUpdating", state.isUpdating - 1)
+        .value();
+    },
+
+    [API_WILL_DELETE]: (state, { payload: resource }) => {
+      const { type, id } = resource;
+
+      return setIsInvalidatingForExistingResource(
+        state,
+        { type, id },
+        IS_DELETING
+      )
+        .set("isDeleting", state.isDeleting + 1)
+        .value();
+    },
+
+    [API_DELETED]: (state, { payload: resource }) => {
+      return removeResourceFromState(state, resource)
+        .set("isDeleting", state.isDeleting - 1)
+        .value();
+    },
+
+    [API_DELETE_FAILED]: (state, { payload: { resource } }) => {
+      const { type, id } = resource;
+
+      return setIsInvalidatingForExistingResource(
+        state,
+        { type, id },
+        IS_DELETING
+      )
+        .set("isDeleting", state.isDeleting - 1)
+        .value();
+    },
+    [API_RESET_RESOURCE]: (state, { payload }) => {
```

Comment:
On a pas vraiment besoin du WILL_XXX and _FAILED our _SUCCESS non ?
C'est pour ça qu'on a qu'une action, comme on fait d'appel API mais qu'on fait juste un reset du state, on a pas besoin,rigth ?

Reply:
Yes, pas de call donc pas besoin des deux, on manipule juste le state